src/components/Text/index.android.js

```diff
@@ -0,0 +1,40 @@
+import React, {forwardRef, useState} from 'react';
+import BaseText from './BaseText';
+import {defaultProps, propTypes} from './baseTextPropTypes';
+
+const Text = (props) => {
+    const [truncatedText, setTruncatedText] = useState(typeof props.children === 'string' ? props.children : null);
+
+    const handleTextLayout = ({nativeEvent: {lines}}) => {
+        if (!props.numberOfLines || lines.length <= props.numberOfLines) {
+            return;
+        }
+
+        let newText = '';
+        for (let index = 0; index < props.numberOfLines; index += 1) {
+            newText = `${newText}${lines[index].text}`;
+        }
+
+        setTruncatedText(`${newText.slice(0, -3)}...`);
+    };
+
+    return (
+        <BaseText
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+            numberOfLines={null}
```

Comment:
I assume `numberOfLines` is always overwritten to null?

Reply:
Yes, in this case, for Android, it needs to be because we are doing the truncating logic in the component itself